modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/GiveRatingViewModel.kt

```diff
@@ -85,38 +79,46 @@ class GiveRatingViewModel @Inject constructor(
                     if (countPlayedEpisodes < NUMBER_OF_EPISODES_LISTENED_REQUIRED_TO_RATE) {
                         _state.value = State.NotAllowedToRate(podcastUuid)
                     } else {
-                        val podcast = podcastManager.findPodcastByUuidSuspend(podcastUuid)
-                        if (podcast == null) {
-                            _state.value = State.ErrorWhenLoadingPodcast
-                        } else {
-                            val rating = ratingsManager.podcastRatings(podcastUuid).first()
-                            val stars: Stars? = getStarsFromRating(rating)
-
-                            _state.value = State.Loaded(
-                                podcastUuid = podcast.uuid,
-                                podcastTitle = podcast.title,
-                                _stars = stars,
-                            )
-                        }
+                        onSuccess()
                     }
                 }
             }
         }
     }
 
-    private fun getStarsFromRating(podcastRatings: PodcastRatings): Stars? {
-        val rating = podcastRatings.average ?: 0.0
-
-        val ratingInt = rating.toInt()
-        val half = rating % 1 >= 0.5
-
-        return when (ratingInt) {
-            0 -> Stars.Zero
-            1 -> if (half) Stars.OneAndHalf else Stars.One
-            2 -> if (half) Stars.TwoAndHalf else Stars.Two
-            3 -> if (half) Stars.ThreeAndHalf else Stars.Three
-            4 -> if (half) Stars.FourAndHalf else Stars.Four
-            else -> Stars.Five // if the rating is somehow higher than 5, just return 5
+    fun loadData(podcastUuid: String) {
+        viewModelScope.launch {
+            _state.value = State.Loading
+
+            val podcast = podcastManager.findPodcastByUuidSuspend(podcastUuid)
+
+            if (podcast == null) {
+                _state.value = State.ErrorWhenLoadingPodcast
+            } else {
+                try {
+                    val ratedPodcast = syncManager.getPodcastRating(PodcastRatingShowRequest(podcastUuid))
```

Comment:
This is just an idea, but I wonder if it would be nicer to move the request creation into the `syncManager` or a `RatingsManager` and pass in the podcast UUID. In a similar way, could it return the rating Int? It could also handle the 404 response code instead of having it in the ViewModel.
```suggestion
                    val rating = syncManager.getPodcastRating(podcastUuid)
```

Reply:
Great idea. I refactor the code and you can see here: [39d2d2c](https://github.com/Automattic/pocket-casts-android/pull/2483/commits/39d2d2cbc2a7a2c69b2b329f0b1f90a09a40615c)